scripts/js/commands/checkNewPills.ts

```diff
@@ -0,0 +1,99 @@
+// This code is a Qiskit project.
+//
+// (C) Copyright IBM 2025.
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+import fs from "fs/promises";
+import path from "path";
+
+import { globby } from "globby";
+import yargs from "yargs/yargs";
+import { hideBin } from "yargs/helpers";
+import { flatten, flattenDeep } from "lodash-es";
+
+import { TocEntry } from "../lib/api/generateToc.js";
+
+const TODAY = new Date();
+
+interface NewPillEntry {
+  url: string;
+  date?: string | "null";
+}
+
+// Grab the toc and grab each url with a newpill
+async function main() {
+  const tocFiles = await globby([
+    "docs/guides/_toc.json",
+    "docs/migration-guides/_toc.json",
+    "docs/open-source/_toc.json",
+  ]);
+  const allOutdatedPills = [];
+  for (const tocFile of tocFiles) {
+    const tocNewPills = await findNewPills(tocFile);
+    const outdatedPills = await getOutdatedPills(tocNewPills);
+    allOutdatedPills.push(...outdatedPills);
+  }
+  if (allOutdatedPills.length > 0) {
+    allOutdatedPills.forEach((outdatedPill: NewPillEntry) =>
+      console.error(
+        `\n❌ This URL needs to have their new pill removed: ${outdatedPill.url}`,
+      ),
+    );
+  } else {
+    console.log("\nNo outdated new pills found ✅\n");
+  }
+}
+
+async function getOutdatedPills(
```

Comment:
This doesn't need to be async because you don't `await` anything.

I'd also recommend calling it something like `detectBadNewPills` to make more clear it's a "pure function" that doesn't do any side effects like reading from the filesystem.

Finally, this function is conflating two different error paths:
1. Outdated pills
2. Missing dates

Those are distinct errors and should have different error messages. So, I recommend having this function return the type `{outdated: NewPillEntry[]; missingDate: NewPillEntry[]}`. In your `main` function, you'll have two arrays for `allOutdatedPills` and `allMissingDates`. They each get their own error message